core/src/main/java/com/scalar/db/exception/transaction/TransactionException.java

```diff
@@ -1,13 +1,44 @@
 package com.scalar.db.exception.transaction;
 
+import java.util.Optional;
+import javax.annotation.Nullable;
+
 /** Base class for all exceptions thrown by the Transaction API. */
 public class TransactionException extends Exception {
 
-  public TransactionException(String message) {
```

Comment:
@brfrn169 Sorry, it might be a stupid question, but is this change backward-compatible? Or is the usage like the following uncommon or unexpected?

https://github.com/scalar-labs/scalardb-benchmarks/blob/b236c1f75dfb5e95337e973a84fe26c38dfd15b8/src/main/java/com/scalar/db/benchmarks/tpcc/transaction/NewOrderTransaction.java#L169

Reply:
@jnmt (CC: @feeblefakie)

The PR changed the signatures of the constructors of the transaction-related exceptions, so applications that instantiate the exceptions will encounter a compile error due to the change. In that sense, it's not backward-compatible.

Currently, ScalarDB doesn't have an explicit rule about which classes/methods/constructors are considered as public (exposed to users) or private (internal use). Mature OSS basically has such a rule, and the developers can make incompatible changes to the private ones, but not to the public ones.

And I thought we didn't assume that end users would use the constructors of the exceptions (so the constructors are considered as private (internal use)), so I made the (backward-incompatible) changes.